mlflow/pyfunc/context.py

```diff
@@ -55,3 +55,16 @@ def get_prediction_context() -> Optional[Context]:
         The context for the current prediction request, or None if no context is set.
     """
     return _PREDICTION_REQUEST_CTX.get()
+
+
+@contextlib.contextmanager
+def maybe_set_prediction_context(context: Optional[Context]):
+    """
+    Only override the prediction context if it is provided. Otherwise, use the existing
+    prediction context or no prediction context at all.
```

Comment:
```suggestion
    Set the prediction context if the given context argument is not None. Otherwise no-op.
```

nit: The diff from `set_prediction-context` is not whether or not to override, rather the behavior when `None` is passed. Some confusion might have come from the term "override". WDYT?